packages/@aws-cdk/aws-ecr-assets/lib/tarball-asset.ts

```diff
@@ -0,0 +1,85 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ecr from '@aws-cdk/aws-ecr';
+import { AssetStaging, Stack, Stage } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line
+import { IAsset } from '@aws-cdk/assets';
+// keep this import separate from other imports to reduce chance for merge conflicts with v2-main
+// eslint-disable-next-line no-duplicate-imports, import/order
+import { Construct as CoreConstruct } from '@aws-cdk/core';
+
+/**
+ * Options for TarballImageAsset
+ */
+export interface TarballImageAssetProps {
+  /**
+   * Path to the tarball (relative to the directory).
+   */
+  readonly tarballFile: string;
+}
+
+/**
+ * An asset that represents a Docker image.
+ *
+ * The image will loaded from an existing tarball and uploaded to an ECR repository.
+ */
+export class TarballImageAsset extends CoreConstruct implements IAsset {
+  /**
+   * The full URI of the image (including a tag). Use this reference to pull
+   * the asset.
+   */
+  public imageUri: string;
+
+  /**
+   * Repository where the image is stored
+   */
+  public repository: ecr.IRepository;
+
+  /**
+   * A hash of the source of this asset, which is available at construction time. As this is a plain
+   * string, it can be used in construct IDs in order to enforce creation of a new resource when
+   * the content hash has changed.
+   * @deprecated use assetHash
```

Comment:
Remove this property...

Reply:
Should I also remove the IAsset interface?

Reply:
Oh I see. It's coming from there... ok leave it as is. It will be automatically removed in v2